lib/protocol/collection/item_collection_notes_view.ts

```diff
@@ -78,13 +76,12 @@ export class ItemCollectionNotesView {
         notes
       );
     } else if (tag) {
-      this.displayedList = this.collection.referencesForElement(tag)
-        .filter(element =>
-          element.content_type === ContentType.Note &&
-          !element.deleted &&
-          !element.trashed &&
-          this.displayFilter(element)
-      ) as SNNote[];
+      const taggedNoteUuids = new Set(
+        this.collection.uuidReferencesForUuid(tag.uuid)
+      );
+      this.displayedList = notes.filter(
+        note => taggedNoteUuids.has(note.uuid)
```

Comment:
You could also just have used `tag.hasRelationshipWithItem(note)`

Reply:
`tag.hasRelationshipWithItem(note)` will search its references list every time, so it's fine when you're checking a single note but probably not when you don't know how many notes you are checking

Reply:
It's filtering through an array either way no?